hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleEntityTracker.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * -
+ *
+ * Hedera Services Node
+ *
+ * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.services.state.merkle;
+
+import com.google.common.base.MoreObjects;
+import com.swirlds.common.io.SelfSerializable;
+import com.swirlds.common.io.SerializableDataInputStream;
+import com.swirlds.common.io.SerializableDataOutputStream;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class MerkleEntityTracker implements SelfSerializable {
+
+	public static final int MERKLE_VERSION = 1;
+
+	public static final long RUNTIME_CONSTRUCTABLE_ID = 0x37f21b97b48c65d1L;
+
+	private long lastScannedAccount;
+
+	private long lastScannedToken;
+
+	public MerkleEntityTracker() {
+	}
+
+	public MerkleEntityTracker(long lastScannedAccount, long lastScannedToken) {
+		this.lastScannedAccount = lastScannedAccount;
+		this.lastScannedToken = lastScannedToken;
+	}
+
+	@Override
+	public int getVersion() {
+		return MERKLE_VERSION;
+	}
+
+	@Override
+	public long getClassId() {
+		return RUNTIME_CONSTRUCTABLE_ID;
+	}
+
+	@Override
+	public void deserialize(SerializableDataInputStream in, int version) throws IOException {
+		lastScannedAccount = in.readLong();
+		lastScannedToken = in.readLong();
+	}
+
+	@Override
+	public void serialize(SerializableDataOutputStream out) throws IOException {
+		out.writeLong(lastScannedAccount);
+		out.writeLong(lastScannedToken);
+	}
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) {
+			return true;
+		}
+		if (o == null || getClass() != o.getClass()) {
+			return false;
+		}
+		var that = (MerkleEntityTracker) o;
+
+		return lastScannedAccount == that.lastScannedAccount &&
+				lastScannedToken == that.lastScannedToken;
+	}
+
+	@Override
+	public int hashCode() {
+		int result = Integer.hashCode(MERKLE_VERSION);
+		result = result * 31 + Long.hashCode(RUNTIME_CONSTRUCTABLE_ID);
```

Comment:
I think we don't need `RUNTIME_CONSTRUCTABLE_ID` in the `hashCode` implementation. Both `hashCode` and `equals` methods should use the same fields.

Reply:
Same goes for `MERKLE_VERSION`